### PR TITLE
feat: add new chatbot interface (graasp/graasp#641)

### PR DESCRIPTION
**Pull request by ReidyT**
This interface represents the new OpenAI v3.5/4 message and it is used in the code capsule app and in the core backend.

This pull request is associated to the graasp/graasp#641 issue.

### CODE REVIEW

File: src/interfaces/chat-bot.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+export interface ChatBotMessage {
+  role: 'system' | 'assistant' | 'user';
```
Comment (spaenleh) [inline src/interfaces/chat-bot.ts:2]: Maybe we want to expose an enum with these values, so it is easier to construct messages without having to remember the exact strings:
```ts
export enum ChatbotRole {
  System = "system",
  Assistant = "assistant",
  User = "user
}
```
And you can get the union of enumerated values like:
```suggestion
  role: `${ChatbotRole}` | ChatbotRole;
```

**Review by spaenleh — CHANGES_REQUESTED** [review]
Looks good to me, I have suggested to add an enum to make creating these messages easier. Let me know what you think ⚙️

File: src/interfaces/chat-bot.ts (hunk 1 above)
Comment (ReidyT) [reply src/interfaces/chat-bot.ts]: That's a good idea !

**Review by spaenleh — APPROVED** [review]
🤖

**Pull request merged**